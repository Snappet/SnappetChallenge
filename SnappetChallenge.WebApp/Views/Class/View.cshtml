
@{
    ViewBag.Title = "Error";
}


<div>An, error has occured and the techncial team was notified. Please, try again shortly.</div>
