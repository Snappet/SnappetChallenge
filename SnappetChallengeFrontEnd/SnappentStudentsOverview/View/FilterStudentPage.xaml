<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
             xmlns:Views="clr-namespace:SnappetChallenge.Client.View"
             xmlns:buttons="clr-namespace:Syncfusion.XForms.Buttons;assembly=Syncfusion.Buttons.XForms"
             xmlns:syncfusion="clr-namespace:Syncfusion.ListView.XForms;assembly=Syncfusion.SfListView.XForms" 
             xmlns:converter="clr-namespace:SnappetChallenge.Client.Control"
             xmlns:viewModels="clr-namespace:SnappetChallenge.Client.ViewModel"
             xmlns:combobox="clr-namespace:Syncfusion.XForms.ComboBox;assembly=Syncfusion.SfComboBox.XForms"
             x:Class="SnappetChallenge.Client.View.FilterStudentPage">
    <ContentPage.Resources>
        <converter:CollectionHasElementsConverter x:Key="CollectionHasElementsConverter"/>
        <converter:IsEqualConverter x:Key="IsEqualConverter"/>
    </ContentPage.Resources>
    <ContentPage.Content>
        <Grid ColumnDefinitions="2*,8*">
            <StackLayout Grid.Column="0">
                <Button Command="{Binding GetFilterStudentItemsCommand}"
                    Text="Get Student By Filter"/>
                <Label Text="Start time"/>
                <DatePicker Date="{Binding FilterStartSpecificTime}"
                             MaximumDate="03/24/2015" />

                <Label Text="End time"/>
                <DatePicker Date="{Binding FilterEndSpecificTime}" 
                             MaximumDate="03/24/2015"/>
                
                <Label Text="Select Domain or multi domains for filter student"/>
                <syncfusion:SfListView 
                            SelectionMode="Multiple"
                            SelectionGesture="Tap"
                            SelectedItems ="{Binding SelectedDomains}"
                            ItemsSource="{Binding StudentDomains}"
                            ItemSize="40">
                    <syncfusion:SfListView.ItemTemplate>
                        <DataTemplate>
                            <Grid Padding="10">
                                <Label Text="{Binding .}" 
                                   FontAttributes="Bold" 
                                   TextColor="Black" 
                                   FontSize="15" />
                            </Grid>
                        </DataTemplate>
                    </syncfusion:SfListView.ItemTemplate>
                </syncfusion:SfListView>
                <BoxView  HeightRequest="5"
                          BackgroundColor="Black"
                          HorizontalOptions="FillAndExpand"/>
                <Label Text="Select Subject or multi Subjects for filter student  "/>
                <syncfusion:SfListView  
                            SelectionMode="Multiple"
                            SelectionGesture="Tap"
                            SelectedItems="{Binding SelectedSubjects}"
                            ItemsSource="{Binding StudentSubjects}"
                            ItemSize="40">
                    <syncfusion:SfListView.ItemTemplate>
                        <DataTemplate>
                            <Grid Padding="10">
                                <Label Text="{Binding .}" 
                                   FontAttributes="Bold" 
                                   TextColor="Black" 
                                   FontSize="15" />
                            </Grid>
                        </DataTemplate>
                    </syncfusion:SfListView.ItemTemplate>
                </syncfusion:SfListView>
               
            </StackLayout>
            <StackLayout Grid.Column="1">
                <Views:ItemLoaderView 
                        IsVisible="{Binding IsStudentListLoading}"/>
                <Grid  
                    IsVisible="{Binding IsStudentListLoading,Converter={StaticResource IsEqualConverter},ConverterParameter=True?False:True}">
                        <StackLayout    
                            IsVisible="{Binding CustomStudentReport, Converter={StaticResource CollectionHasElementsConverter}}"
                            Spacing="0">
                            <Label 
                                Margin="15,15,15,0"
                                HorizontalTextAlignment="Center"
                                Text="There are no items" />
                            <buttons:SfButton
                                    HorizontalOptions="Center"
                                    BackgroundColor="White"
                                    Command="{Binding ClearFilterCommand}">
                                <buttons:SfButton.Content>
                                    <Label  Text="Clear filter"
                                        TextDecorations="Underline"/>
                                </buttons:SfButton.Content>
                            </buttons:SfButton>
                        </StackLayout>

                    <syncfusion:SfListView  
                                    AutoFitMode="DynamicHeight"
                                    BackgroundColor="White"
                                    x:Name="StudentList"
                                    VerticalOptions="FillAndExpand"
                                    IsVisible="{Binding CustomStudentReport, Converter={StaticResource CollectionHasElementsConverter}}"
                                    ItemsSource="{Binding Source={RelativeSource AncestorType={x:Type viewModels:FilterStudentViewModel}}, Path=CustomStudentReport}"
                                    SelectionMode="Single"
                                    ItemSize="60"
                                    FocusBorderColor="White"
                                    SelectionBackgroundColor="Transparent"
                                    LoadMoreCommand="{Binding LoadMoreStudentApplyFilterCommand}"
                                    LoadMoreCommandParameter="{Binding Source={x:Reference Name=StudentList}}"
                                    LoadMoreOption="AutoOnScroll"
                                    IsBusy="{Binding IsStudentListLoading}"
                                    TapCommand="{Binding SelectedMeetingCommand,ConverterParameter={Binding .}}">
                        <syncfusion:SfListView.ItemTemplate>
                            <DataTemplate>
                                <Grid Padding="10"
                                      RowDefinitions="*,*,2"
                                      ColumnDefinitions="2*,8*">
                                    <StackLayout 
                                        Grid.Row="0"
                                        Grid.Column="0">
                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="UserId" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding UserId}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="ExerciseId" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding ExerciseId}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                        
                                    </StackLayout>

                                    <StackLayout
                                         Grid.Row="1"
                                        Grid.Column="0">
                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="SubmittedAnswerId" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding SubmittedAnswerId}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="Correct" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding Correct}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                    </StackLayout>
                                    <StackLayout
                                        Grid.Row="0"
                                        Grid.Column="1"
                                        HorizontalOptions="Start">
                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="Difficulty" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding Difficulty}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>

                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="Subject" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding Subject}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                    </StackLayout>
                                    <StackLayout
                                         Grid.Row="1"
                                        Grid.Column="1"
                                         HorizontalOptions="Start">

                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="Domain" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding Domain}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>

                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="LearningObjective" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding LearningObjective}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                        <Label  FontAttributes="Bold" 
                                               TextColor="Teal" FontSize="21" >
                                            <Label.FormattedText>
                                                <FormattedString>
                                                    <Span Text="SubmitDateTime" />
                                                    <Span Text="  " />
                                                    <Span Text="{Binding SubmitDateTime}"/>
                                                </FormattedString>
                                            </Label.FormattedText>
                                        </Label>
                                    </StackLayout>
                                    <BoxView Grid.Row="2"
                                             Grid.ColumnSpan="2"
                                            HeightRequest="2"
                                             BackgroundColor="Black"/>
                                </Grid>
                            </DataTemplate>
                        </syncfusion:SfListView.ItemTemplate>
                        <syncfusion:SfListView.LoadMoreTemplate>
                            <DataTemplate>
                                <Grid HeightRequest="80"
                                        IsVisible="{Binding IsStudentListLoading}">
                                    <syncfusion:LoadMoreIndicator
                                            Color="LightGray" 
                                            IsRunning="{Binding IsBusy, Source={x:Reference Name=StudentList}}" 
                                            VerticalOptions="Center"/>
                                </Grid>
                            </DataTemplate>
                        </syncfusion:SfListView.LoadMoreTemplate>
                    </syncfusion:SfListView>    
                </Grid>
            </StackLayout>
        </Grid>
        
        
        
        
    </ContentPage.Content>
</ContentPage>