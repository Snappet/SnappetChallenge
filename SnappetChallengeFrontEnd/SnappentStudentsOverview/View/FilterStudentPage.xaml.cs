using SnappetChallenge.Client.ViewModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Xamarin.Forms;
using Xamarin.Forms.Xaml;

namespace SnappetChallenge.Client.View
{
    [XamlCompilation(XamlCompilationOptions.Compile)]
    public partial class FilterStudentPage : ContentPage
    {
        FilterStudentViewModel viewModel;
        public FilterStudentPage()
        {
            InitializeComponent();
            viewModel = new FilterStudentViewModel();
            BindingContext = viewModel;
        }

        private void GetStudentByFilter(object sender, EventArgs e)
        {

        }
        protected override async void OnAppearing()
        {
            base.OnAppearing();
            await viewModel.LoadDomains();
            await  viewModel.LoadSubjects();
        }
    }
}