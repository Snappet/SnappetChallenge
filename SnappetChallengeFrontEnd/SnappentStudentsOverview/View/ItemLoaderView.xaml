<?xml version="1.0" encoding="utf-8" ?>
<ContentView xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
             xmlns:control="clr-namespace:SnappetChallenge.Client.Control"
             x:Class="SnappetChallenge.Client.View.ItemLoaderView">
    <ContentView.Content>
        <StackLayout Spacing="0">
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
            <Grid RowSpacing="0"
                      ColumnSpacing="10"
                      Padding="10,10,10,10"
                      ColumnDefinitions="60,*"
                      HeightRequest="60">

                <control:SkeletonView
                                Grid.Column="0"                        
                                Grid.Row="0"
                                HeightRequest="60"
                                WidthRequest="60"                        
                                VerticalOptions="Start"
                                HorizontalOptions="Center"/>
                <Grid   
                        Grid.Column="1"
                        Grid.Row="0"
                        VerticalOptions="Start"                        
                        RowDefinitions="*,0.8*"
                        RowSpacing="5">

                    <control:SkeletonView 
                            Grid.Row="0"
                            VerticalOptions="Start"/>

                    <control:SkeletonView  
                            Grid.Row="1"
                            VerticalOptions="Start"/>

                </Grid>
            </Grid>
        </StackLayout>
    </ContentView.Content>

</ContentView>