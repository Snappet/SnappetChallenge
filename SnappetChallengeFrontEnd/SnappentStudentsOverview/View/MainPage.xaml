<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="SnappetChallenge.View.MainPage">
    <StackLayout HorizontalOptions="Center" VerticalOptions="Center" Padding="10">
        <Button Text="Open StatisticalReport" Clicked="OpenStatisticalReportClikced"/>
        <Button Text="Open Filter student" Clicked="OpenFilterStudentClikced"/>
        <StackLayout Orientation="Horizontal">
            <Label Text="Change the switch to work on developmet environment http://localhost:5226/api/Student"/>
            <Switch IsToggled="{Binding IsDevelopmetEnvironment}" />
        </StackLayout>
        
    </StackLayout>

</ContentPage>