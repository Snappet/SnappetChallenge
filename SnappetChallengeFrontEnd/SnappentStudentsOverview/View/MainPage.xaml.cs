using SnappetChallenge.Client.Model;
using SnappetChallenge.Client.View;
using SnappetChallenge.Client.ViewModel;
using System;
using System.Threading.Tasks;
using Xamarin.Forms;

namespace SnappetChallenge.View
{
    public partial class MainPage : ContentPage
    {
        BaseViewModel viewModel;

        public MainPage()
        {
            InitializeComponent();
            viewModel = new BaseViewModel();
            BindingContext = viewModel;
        }
        private async void OpenStatisticalReportClikced(object sender, EventArgs e)
        {
            await Application.Current.MainPage.Navigation.PushAsync(new TodayStatisticalReportPage(), false);
        }

        private async void OpenFilterStudentClikced(object sender, EventArgs e)
        {
            await Application.Current.MainPage.Navigation.PushAsync(new FilterStudentPage(), false);
        }
    }
}
    

