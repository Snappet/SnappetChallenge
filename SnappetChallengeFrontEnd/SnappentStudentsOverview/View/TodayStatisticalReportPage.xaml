<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="SnappetChallenge.Client.View.TodayStatisticalReportPage">
    <ContentPage.Content>
        <StackLayout>
            <Button Clicked="LoadTodayStatisticalReport" Text="Load Today Statistical Report"/>
            <ListView x:Name="lstStudentReport" 
                    HorizontalOptions="StartAndExpand"
                    VerticalOptions="FillAndExpand"
                    SeparatorColor="LightGray" 
                    SeparatorVisibility="Default" 
                    HasUnevenRows="True"
                    ItemsSource="{Binding StatisticalReport}" 
                    CachingStrategy="RecycleElement">
                <ListView.ItemTemplate>
                    <DataTemplate>
                        <ViewCell>
                            <StackLayout Orientation="Horizontal" Padding="5" VerticalOptions="FillAndExpand"
                                    MinimumHeightRequest="100">
                                <StackLayout Orientation="Vertical" Padding="2" VerticalOptions="Center" Spacing="2">
                                    <StackLayout Orientation="Horizontal" HorizontalOptions="Start">
                                        <Label Text="Subject" FontSize="12" TextColor="Gray"/>
                                        <Label Text="{Binding Subject}" FontSize="12" TextColor="Gray"/>

                                    </StackLayout>
                                    <StackLayout Orientation="Horizontal" HorizontalOptions="Start">
                                        <Label Text="Correct" FontSize="12" TextColor="Gray"/>
                                        <Label Text="{Binding Correct}" FontSize="12" TextColor="Gray"/>

                                    </StackLayout>

                                    <StackLayout Orientation="Horizontal" HorizontalOptions="Start">
                                        <Label Text="Pass Count" FontSize="12" TextColor="Gray"/>
                                        <Label Text="{Binding PassCount}" FontSize="12" TextColor="Gray"/>

                                    </StackLayout>

                                    <StackLayout Orientation="Horizontal">
                                        <Label Text="Fail Count" FontSize="12" TextColor="#3483eb"/>
                                        <Label Text="{Binding FailCount}" FontSize="12" TextColor="#3483eb"/>
                                    </StackLayout>

                                </StackLayout>
                            </StackLayout>
                        </ViewCell>
                    </DataTemplate>
                </ListView.ItemTemplate>
            </ListView>

        </StackLayout>
    </ContentPage.Content>
</ContentPage>