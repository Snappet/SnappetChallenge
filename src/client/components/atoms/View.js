import styled from 'styled-components';

export default styled.div`
  display: flex;
  flex-direction: ${props => props.flexDirection || 'column'};
  align-items: ${props => props.alignItems || 'center'};
  justify-content: ${props => props.justifyContent || 'center'};
  flex: ${props => props.flex || 'inherit'};
`;
