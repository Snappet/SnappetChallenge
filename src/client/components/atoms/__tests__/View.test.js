import React from 'react';
import renderer from 'react-test-renderer';

import View from '../View';

describe('components', () => {
  describe('atoms', () => {
    describe('View', () => {
      test('should render a View element as a flex div', () => {
        const tree = renderer.create(<View>content</View>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('flex', 'inherit');
      });

      test('should render a View element with flex-start aligned content', () => {
        const tree = renderer.create(
          <View justifyContent="flex-start" alignItems="flex-start">content</View>
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('justify-content', 'flex-start');
        expect(tree).toHaveStyleRule('align-items', 'flex-start');
        expect(tree).toHaveStyleRule('flex', 'inherit');
      });

      test('should render a View element with flex size 5 and direction row', () => {
        const tree = renderer.create(
          <View flex="5" flexDirection="row">content</View>
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('flex', '5');
      });
    });
  });
});
