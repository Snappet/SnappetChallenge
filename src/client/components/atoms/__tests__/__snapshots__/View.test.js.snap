// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components atoms View should render a View element as a flex div 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: inherit;
  -ms-flex: inherit;
  flex: inherit;
}

<div
  className="c0"
>
  content
</div>
`;

exports[`components atoms View should render a View element with flex size 5 and direction row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 5;
  -ms-flex: 5;
  flex: 5;
}

<div
  className="c0"
>
  content
</div>
`;

exports[`components atoms View should render a View element with flex-start aligned content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: inherit;
  -ms-flex: inherit;
  flex: inherit;
}

<div
  className="c0"
>
  content
</div>
`;
